### PR TITLE
sync: move into `tokio` crate

### PR DESCRIPTION
**Pull request by carllerche**
A step towards collapsing Tokio sub crates into a single `tokio`
crate (#1318).

The sync implementation is now provided by the main `tokio` crate.
Functionality can be opted out of by using the various net related
feature flags.

### CODE REVIEW

File: tokio/src/net/unix/incoming.rs [hunk 1]
```diff
@@ -1,5 +1,3 @@
-#![cfg(feature = "async-traits")]
```
Comment (jonhoo) [inline tokio/src/net/unix/incoming.rs]: We no longer want this feature?

File: tokio/src/sync/loom.rs [hunk 2]
```diff
@@ -0,0 +1,48 @@
+#[cfg(not(all(test, loom)))]
+mod imp {
```
Comment (jonhoo) [inline tokio/src/sync/loom.rs]: Do we want to merge this and the `mod loom` used by `tokio::executor` while we're at it?

File: tokio/src/sync/tests/loom_atomic_waker.rs [hunk 3]
```diff
@@ -1,12 +1,4 @@
-#![warn(rust_2018_idioms)]
-
-#[macro_use]
-extern crate loom;
-
-#[allow(dead_code)]
-#[path = "../src/task/atomic_waker.rs"]
-mod atomic_waker;
-use crate::atomic_waker::AtomicWaker;
+use crate::sync::task::AtomicWaker;
```
Comment (jonhoo) [inline tokio/src/sync/tests/loom_atomic_waker.rs]: Not sure I follow this — shouldn't it now be `crate::task::AtomicWaker`?

File: tokio/src/sync/tests/loom_list.rs [hunk 4]
```diff
@@ -1,3 +1,4 @@
+#![cfg(broken)]
```
Comment (jonhoo) [inline tokio/src/sync/tests/loom_list.rs]: Err, what?

File: tokio/src/sync/tests/loom_list.rs [hunk 5]
```diff
@@ -8,6 +9,7 @@ macro_rules! if_fuzz {
         $($t)*
     }
 }
+/*
```
Comment (jonhoo) [inline tokio/src/sync/tests/loom_list.rs]: What's the reason for this?

File: tokio/src/sync/loom.rs (hunk 2 above)
Comment (hawkw) [reply tokio/src/sync/loom.rs]: I think the intention is to consolidate this (and also the `mod loom` in `tokio-net`) in a follow-up PR.

File: tokio/src/net/unix/incoming.rs (hunk 1 above)
Comment (carllerche) [reply tokio/src/net/unix/incoming.rs]: I plan on going back and doing a full sweep of feature flags.

File: tokio/src/sync/loom.rs (hunk 2 above)
Comment (carllerche) [reply tokio/src/sync/loom.rs]: follow up PR. I'm trying to "just" collapse crates here. There is a lot of follow up cleanup to do.

File: tokio/src/sync/tests/loom_list.rs (hunk 4 above)
Comment (carllerche) [reply tokio/src/sync/tests/loom_list.rs]: woops!

File: tokio/src/sync/tests/loom_list.rs (hunk 5 above)
Comment (carllerche) [reply tokio/src/sync/tests/loom_list.rs]: Mistake on my part. Fixed.

Comment (carllerche) [reply tokio/src/sync/tests/loom_list.rs]: Oh, we don't need the mod / path hacks anymore.

**Pull request merged**